src/cpu/callback.cpp

```diff
@@ -195,379 +212,315 @@ static uint8_t callback_setup_extra(const callback_number_t callback_number,
 		if (should_use_callback) {
 			add_native_call_4(callback_number);
 		}
-		add_instruction_1(0xc3); // RETN
+		add_instruction_1(0xc3);                   // retn
 		break;
 	case CB_RETF:
 		if (should_use_callback) {
 			add_native_call_4(callback_number);
 		}
-		add_instruction_1(0xcb); // RETF
+		add_instruction_1(0xcb);                   // retf
 		break;
 	case CB_RETF8:
-		if (use_cb) {
-			phys_writeb(physAddress+0x00,(uint8_t)0xFE);	//GRP 4
-			phys_writeb(physAddress+0x01,(uint8_t)0x38);	//Extra Callback instruction
-			phys_writew(physAddress + 0x02, cb_num);        // The immediate word
-			physAddress += 4;
+		if (should_use_callback) {
+			add_native_call_4(callback_number);
 		}
-		phys_writeb(physAddress+0x00,(uint8_t)0xCA);		//A RETF 8 Instruction
-		phys_writew(physAddress+0x01,(uint16_t)0x0008);
-		return (use_cb?7:3);
+		add_instruction_3(0xca, 0x08, 0x00);       // retf 8
```

Comment:
Oh man, manually returning and maintaining all these values... Why do people do such things to themselves... 😕

Reply:
I can't understand this neither. True, they didn't have lambdas back then - but passing a reference to current address variable would have improved this, too.